crates/transaction-pool/src/pool/mod.rs

```diff
@@ -407,8 +419,10 @@ where
 
                 // transaction was successfully inserted into the pool
                 if let Some(sidecar) = maybe_sidecar {
-                    // store the sidecar in the blob store
-                    self.insert_blob(hash, sidecar);
+                    // store the (cloned) sidecar in the blob store
+                    self.insert_blob(hash, sidecar.clone());
+                    // notify blob sidecar listeners
+                    self.on_new_blob_sidecar(sidecar)
```

Comment:
this is a bit problematic, we don't expect to have listeners by default but we now clone always,

the order should be switched and on_new_blob should take a ref

Reply:
absolutely, will fix!

I felt a little sleazy cloning it here, hence the comment haha